### PR TITLE
[Platform][RBLN] MLA (NL_X_NB_BS_HS) block transfer support

### PR DESCRIPTION
**Pull request by rebel-jindol21**
**What this PR does / why we need it**:

Adds MLA KV-cache support to the RBLN block-transfer path. vLLM-RBLN's MLA attention backend allocates each layer as `[NB, BS, HS]` (a single latent plane, no K/V split and no head axis), which the vLLM detector classifies as `EngineKVFormat.NL_X_NB_BS_HS`. `RblnDeviceOps` previously rejected everything but the HND attention layout, so MLA models could not use KV offloading on RBLN.

- `RblnDeviceOps.multi_layer_block_kv_transfer` now accepts the MLA format alongside HND, classifying it with `lmcache_native.is_mla()` and dispatching to an MLA-specific op sequence; the chunk/block bookkeeping stays shared. MLA layouts other than `NL_X_NB_BS_HS` are rejected at the format check with a format error.
- The MLA gather/scatter (`gather_blocks_to_chunk_mla` / `scatter_chunk_to_blocks_mla`) is built functionally (`index_select` + `stack`, `index_copy_`) so it stays on device-native v2v kernels and crosses the device boundary with one DMA per chunk. The shared torch path stages through `torch.empty(device=...)`, which on RBLN is a lazy SHM tensor that drops the ops onto the CPU-fallback path.
- `validate_mla_layers` pins the 3-D rank at the transfer boundary so a layout drift fails loudly instead of mis-addressing bytes.
- The existing HND kernels are renamed `gather_blocks_to_chunk_hnd` / `scatter_chunk_to_blocks_hnd` to mirror the `_mla` pair.

Chunks keep LMCache's canonical token-major wire layout (`[L, T, HS]` for MLA), so they remain byte-compatible with chunks produced on other devices. The design rationale is documented in `docs/design/v1/platform/rbln/README.md`.

**Special notes for your reviewers**:

- Both layouts require the engine's KV caches to be real device tensors (`VLLM_RBLN_USE_DEVICE_TENSOR=1`); with compile-mode allocation the per-layer tensors are `meta` and any transfer dies at the first host copy.

**If applicable**:

- [x] this PR contains user facing changes - docs added
- [x] this PR contains unit tests

### CODE REVIEW

**Comment by rebel-jindol21** [comment]
@maobaolong @DongDongJu @hlin99 Hello, I am working with @rebel-jinhwan in Rebellions. This feature supports MLA in our devices. For MLA, most of works come from this PR(https://github.com/LMCache/LMCache/pull/4450). Please leave comments on this work. I will look into it.

cc. @rebel-jinhwan

File: lmcache/v1/platform/rbln/kv_ops.py [hunk 1]
```diff
@@ -154,3 +154,77 @@ def scatter_chunk_to_blocks(
...
+    Args:
+        paged_layers: Per-layer MLA KV tensors, each ``[NB, BS, HS]``.
+        block_ids: Blocks to gather, in chunk-token order.
+        dst: Chunk shaped ``[L, T, HS]``. Only its leading
+            ``len(block_ids) * BS`` tokens are written, so a trailing chunk
+            holding fewer blocks than it was sized for is fine.
+    """
+    n_blocks = len(block_ids)
+    _nb, block_size, head_size = paged_layers[0].shape
+    idx = torch.as_tensor(block_ids, dtype=torch.long, device=paged_layers[0].device)
```
Comment (DongDongJu) [inline lmcache/v1/platform/rbln/kv_ops.py]: idx is created on the RBLN device and reused by index_select once per layer. 
However, torch-rbln v0.3.0’s [read_index_to_host()](https://github.com/RBLN-SW/torch-rbln/blob/v0.3.0/aten/src/ATen/native/rbln/RBLNIndexUtils.h#L20-L38) immediately calls .cpu() for every device index. 
Since RBLN index_select accepts a CPU index, this gather index should remain on CPU.

File: lmcache/v1/platform/rbln/kv_ops.py [hunk 2]
```diff
@@ -154,3 +154,77 @@ def scatter_chunk_to_blocks(
...
+    # [L, B, BS, HS] on the paged tensors' device; stack decomposes to the
+    # v2v cat kernel, and the result is contiguous so the view below is free.
+    gathered = torch.stack(
+        [torch.index_select(layer, 0, idx) for layer in paged_layers]
+    )
+    dst[:, : n_blocks * block_size].copy_(
+        gathered.view(len(paged_layers), n_blocks * block_size, head_size)
```
Comment (DongDongJu) [inline lmcache/v1/platform/rbln/kv_ops.py]: So, Scatter cannot make the same simple change because index_copy requires the index to share the destination device, but each layer still materializes that index on the host. 
More importantly, torch-rbln’s [submit_or_fallback](https://github.com/RBLN-SW/torch-rbln/blob/v0.3.0/aten/src/ATen/native/rbln/RBLNStridedV2V.cpp#L118-L149) explicitly falls back through CPU when V2V rejects interior offsets in large untyped allocations such as vLLM KV caches. 
That makes the current “device-native” and “one DMA per chunk” claims unproven.

Before merging, please check(share to us is even better) a trace from an actual vLLM-RBLN MLA KV cache using VLLM_RBLN_USE_DEVICE_TENSOR=1, including the torch-rbln/vllm-rbln versions and a store→retrieve correctness result. The trace should demonstrate that index_select, stack/cat, and index_copy_ do not take the CPU-fallback path. If they do, please compare a block-view implementation using _foreach_copy_.

Not sure how long we can maintain this with non-official vllm and torch.

**Review by DongDongJu — CHANGES_REQUESTED** [review]
Hello @rebel-jindol21, Thanks for the contribution.
I left one question and request.
PTAL!

File: lmcache/v1/platform/rbln/kv_ops.py (hunk 1 above)
Comment (rebel-jindol21) [reply lmcache/v1/platform/rbln/kv_ops.py]: Thanks for kind review. Your view is exactly right, and the read-back is per layer..
Trace on a bound RBLN KV cache shaped like the MLA one (L=27, NB=1024, BS=128, HS=576, bf16; 2-block chunk = 7.6 MiB; 5 calls, fresh process; torch-rbln 0.4.0rc0 on RBLN-CR03. I checked the v0.3.0 tag you linked against 0.4.0rc0: the indexing/v2v kernels are identical, only RBLNFunctions.cpp differs, so read_index_to_host() behaves the same on both):

| gather index        | v2h calls over the 5 gathers                                                                            |
|------------------|-------------------------------------------------------------------------|
| device (this PR)   | 140 = per call, 27 × read_index_to_host() + 1 chunk D2H                           |
| CPU                      | 5 = per call, 1 chunk D2H                                                                                 |

So index_select does take a CPU index on RBLN, and keeping the index on device costs exactly one extra v2h per layer per call, as you said. Each read-back moves only the index itself, but the call count is unambiguous and scales with layer count.

Rather than moving the index to CPU, I dropped index tensors from the MLA path entirely. Block ids stay as Python ints and every block is copied whole, so v2h is 1 per gather by construction.

File: lmcache/v1/platform/rbln/kv_ops.py (hunk 2 above)
Comment (rebel-jindol21) [reply lmcache/v1/platform/rbln/kv_ops.py]: You were right that the claims were unproven; running the trace changed the implementation.

On the real vLLM-RBLN MLA KV cache (DeepSeek-V3, `VLLM_RBLN_USE_DEVICE_TENSOR=1`),
neither `index_select`/`stack`/`index_copy_` nor a block-view `_foreach_copy_` variant takes the
CPU-fallback path, and store→retrieve is byte-exact for both. I still switched the MLA path to the
block-view `_foreach_copy_` design: no index tensor, no `submit_or_fallback` behind any
op, and 3.5–5× faster once a chunk holds more than one block.

**Setup:** `deepseek-ai/DeepSeek-V3` (61 layers, FP8), our DeepSeek-V3 release-bench serve config
(`vllm serve --block-size 1024 --data-parallel-size 8 --enable-expert-parallel --enable-chunked-prefill
--no-enable-prefix-caching --max-model-len 65536 --max-num-batched-tokens 512 --max-num-seqs 1
--num-gpu-blocks-override 64`), 8× RBLN-CR03, `LMCacheMPConnector` + MP server (engine_driven, chunk 1024).
Versions: torch-rbln 0.4.0rc0 (indexing/v2v/copy kernels identical to v0.3.0), vllm-rbln
0.11.3a1.dev3, rebel-compiler 0.11.3.dev31, vllm 0.24.0.

**CPU fallback:** 0 across all five bounce sites and 0 `cpu_fallback` ops for both implementations

**Correctness:** 16/16 store→retrieve pairs (ISL 1k/2k/4k/8k × both implementations × 2) retrieved with
byte-identical chunk digests; a natural-language prompt gives identical cold/warm output for both
implementations.

On maintenance: I understand what you are worrying about. 
I am afraid that I would say that using our testing system outside is in the middle of development.
We will let you know after we make a resolution.

Comment (rebel-jinhwan) [reply lmcache/v1/platform/rbln/kv_ops.py]: @DongDongJu Thank you for your review. We have updated the PR based on your comments. Could you please review the changes?

Additionally, regarding running Buildkite CI on our infrastructure from the outside, we still need some time for preparation. In the meantime, if there are any specific parts you need verified, please let us know. We can manually run the tests and share the results with you.

Thank you!

**Review by DongDongJu — APPROVED** [review]
Sry I was missed this. Thanks for the contribution.

**Comment by rebel-jinhwan** [comment]
@maobaolong Could you take a look when you have a moment? Thank you!

**Review by maobaolong — APPROVED** [review]
@rebel-jindol21 Since all the addition codes are within the `rbln` module, so it has no affect to the lmcache core. LGTM.

**Comment by maobaolong** [comment]
@rebel-jindol21 Just hope one day we can have a CI pipeline can help to verify every thing work as expected.
